spaceship.zsh

```diff
@@ -293,11 +293,18 @@ _is_git() {
   command git rev-parse --is-inside-work-tree &>/dev/null
 }
 
-# Check if the current direcotory is in a Mercurial repository
+# Check if the current directory is in a Mercurial repository.
 # USAGE:
-#   _is_hg
+#   if [ $(_is_hg) ]
 _is_hg() {
-  command hg --cwd $PWD root &>/dev/null
+  local root="$(pwd -P)"
+
+  while [[ $root && ! -d $root/.hg ]]
+  do
+    root="${root%/*}"
+  done
+
+  echo "$root"
```

Comment:
What is `$root`? Can we return simplify this to use this as `_is_hg()`, without `if`?